js/hitbtc3.js

```diff
@@ -276,100 +276,145 @@ module.exports = class hitbtc3 extends Exchange {
-            const amountIncrement = this.safeNumber (entry, 'quantity_increment');
-            const maxLeverage = this.safeNumber (entry, 'max_initial_leverage');
-            const precision = {
-                'price': priceIncrement,
-                'amount': amountIncrement,
-            };
-            const limits = {
-                'amount': {
-                    'min': undefined,
-                    'max': undefined,
-                },
-                'price': {
-                    'min': undefined,
-                    'max': undefined,
-                },
-                'cost': {
-                    'min': undefined,
-                    'max': undefined,
-                },
-                'leverage': {
-                    'min': minLeverage,
-                    'max': maxLeverage,
-                },
-            };
+            const lotString = this.safeString (market, 'quantity_increment');
+            const stepString = this.safeString (market, 'tick_size');
+            const lot = this.parseNumber (lotString);
+            const step = this.parseNumber (stepString);
+            const taker = this.safeNumber (market, 'take_rate');
+            const maker = this.safeNumber (market, 'make_rate');
+            const fees = this.safeValue (this.fees, 'trading');
             result.push ({
-                'info': entry,
-                'symbol': symbol,
                 'id': id,
+                'symbol': symbol,
                 'base': base,
                 'quote': quote,
+                'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
+                'settleId': settleId,
+                'type': type,
                 'spot': spot,
                 'margin': margin,
-                'futures': futures,
-                'type': type,
-                'feeSide': feeSide,
-                'maker': maker,
+                'swap': swap,
+                'future': future,
+                'futures': future,  // Deprecated, use future instead
+                'option': option,
+                'derivative': derivative,
+                'contract': contract,
+                'linear': linear,
+                'inverse': inverse,
                 'taker': taker,
-                'precision': precision,
-                'limits': limits,
-                'expiry': undefined,
+                'maker': maker,
+                'contractSize': contractSize,
+                'active': true,
+                'expiry': expiry,
                 'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'feeCurrency': feeCurrency,
+                'fees': {
+                    'tierBased': this.safeValue (fees, 'tierBased'),
+                    'percentage': this.safeValue (fees, 'percentage'),
+                    'taker': taker,
+                    'maker': maker,
+                },
```

Comment:
fees

Reply:
We don't do this, cause those properties are automatically populated in the base class' setMarkets method. We should just fill `taker` and `maker` and that's it.